apps/dashboard/app/models/project.rb

```diff
@@ -8,6 +8,29 @@ class Project
   include IconWithUri
   extend JobLogger
 
+  def self.from_directory(dir)
+    # fetch "id" by opening .ondemand/manifest.yml
+    manifest_path = Pathname("#{dir.to_s}/.ondemand/manifest.yml")
+    unless manifest_path.exist?
+      Rails.logger.warn("Imported directory is not a Open OnDemand project")
+      return nil
+    end
+
+    contents = File.read(manifest_path)
+    raw_opts = YAML.safe_load(contents)
+    id = raw_opts["id"]
+
+    project = Project.find(id)
+    if project.nil?
+      Project.new({ id: id, directory: dir })
+    else
+      nil
+    end
```

Comment:
OK - I think this has gotten me twisted around. The fact that at this point the project could be a part of the lookup file already.

Seems like this method `from_directory` shouldn't consult the lookup table. It should only try to create a project from a directory. I think this bit was confusing me - that it's not strictly `create a project object from a directory` it's currently more like `create a project object from a directory unless it's already in the lookup table`.

I think this method shouldn't consult the lookup table and just simply return a Project object from a directory if it can.

Then, `import_to_lookup` can consult the lookup table and deal with the case that it's already imported.

Reply:
Totally agree, let me fix this one.